### PR TITLE
network: reconfigure interface after coming back from sleep

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
Many CIs are red, but networkd-test.py on s390x passed.

**Comment by keszybz** [comment]
Rebase please.

**Comment by yuwata** [comment]
Rebased.

File: src/network/networkd-manager.c [hunk 1]
```diff
@@ -85,7 +85,7 @@ static int match_prepare_for_sleep(sd_bus_message *message, void *userdata, sd_b
         if (b)
                 return 0;
 
-        log_debug("Coming back from suspend, resetting all connections...");
+        log_debug("Coming back from sleep, resetting all connections...");
```
Comment (keszybz) [inline src/network/networkd-manager.c]: I think it's better to keep "suspend" here. "sleep" is the kernel name, but it is very ambiguous. We (and pretty much everybody else) calls this suspend. We have `systemctl suspend`, `suspend-then-hibernate`, etc.

File: src/network/networkd-link.c [hunk 2]
```diff
@@ -1239,23 +1239,22 @@ static int link_get_network(Link *link, Network **ret) {
...
-        if (r < 0)
+        if (r < 0 && r != -ENOENT)
                 return r;
 
         if (link->network == network && !force)
                 return 0;
 
-        log_link_info(link, "Re-configuring with %s", network->filename);
+        if (network)
+                log_link_info(link, "Re-configuring with %s.", network->filename);
```
Comment (keszybz) [inline src/network/networkd-link.c]: Re-configuring → Reconfiguring

**Review by keszybz — COMMENTED** [review]
I don't grok all the state transitions, but the code lgtm.

**Comment by yuwata** [comment]
@keszybz Thank you for the review. The above two points are addressed. Upgrading the green label.

**Pull request merged**